src/Raven.Server/Documents/Handlers/StreamingHandler.cs

```diff
@@ -134,29 +134,56 @@ public async Task StreamQueryPost()
-
-                using (var writer = GetQueryResultWriter(format, HttpContext.Response, context, ResponseBodyStream(), propertiesArray, fileName))
+                var fileName = query.Metadata.IsCollectionQuery ? query.Metadata.CollectionName + "_collection" : "query_result";
+                fileName = $"{Database.Name}_{fileName}";
+                if (string.IsNullOrWhiteSpace(debug) == false)
                 {
-                    try
+                    using (var writer = GetIndexEntriesQueryResultWriter(format, HttpContext.Response, context, ResponseBodyStream(), propertiesArray, fileName))
                     {
-                        await Database.QueryRunner.ExecuteStreamQuery(query, context, HttpContext.Response, writer, token).ConfigureAwait(false);
+                        try
+                        {
+                            await Database.QueryRunner.ExecuteStreamIndexEntriesQuery(query, context, HttpContext.Response, writer, token).ConfigureAwait(false);
+                        }
+                        catch (IndexDoesNotExistException)
+                        {
+                            HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                            writer.WriteError($"Index {query.Metadata.IndexName}  does not exists");
+                        }
                     }
-                    catch (IndexDoesNotExistException)
+                }
+                else
+                {
+                    using (var writer = GetQueryResultWriter(format, HttpContext.Response, context, ResponseBodyStream(), propertiesArray, fileName))
                     {
-                        HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
-                        writer.WriteError("Index " + query.Metadata.IndexName + " does not exists");
+                        try
+                        {
+                            await Database.QueryRunner.ExecuteStreamQuery(query, context, HttpContext.Response, writer, token).ConfigureAwait(false);
+                        }
+                        catch (IndexDoesNotExistException)
+                        {
+                            HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                            writer.WriteError($"Index {query.Metadata.IndexName} does not exists");
+                        }
                     }
                 }
             }
         }
 
-        private IStreamDocumentQueryResultWriter GetQueryResultWriter(string format, HttpResponse response, DocumentsOperationContext context, Stream responseBodyStream,
+        private StreamCsvBlittableQueryResultWriter GetIndexEntriesQueryResultWriter(string format, HttpResponse response, DocumentsOperationContext context, Stream responseBodyStream,
+            string[] propertiesArray, string fileName = null)
+        {
+            if (string.IsNullOrEmpty(format) || format.StartsWith("csv") == false)
```

Comment:
its sent as {format: "csv"} from query.ts, therefore it cant be anything aside "csv".

Reply:
If you send a request directly from a browser:
- streams/queries?debug=entries&format=CSV - will fail (it's case sensitive currently)
- streams/queries?debug=entries&format=csveeeee - won't fail but it should because we don't support `csveeee` format

Reply:
done, but currently going direct to:
- streams/queries?format=csv&debug=entries

throws System.InvalidOperationException